example/boundary_exchange/boundary_exchange_driver.cpp

```diff
@@ -0,0 +1,159 @@
+//========================================================================================
+// (C) (or copyright) 2020-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+// Standard Includes
+#include <fstream>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+// Parthenon Includes
+#include <parthenon/driver.hpp>
+
+// Local Includes
+#include "boundary_exchange.hpp"
+#include "boundary_exchange_driver.hpp"
+#include "mesh/forest/forest_node.hpp"
+#include "mesh/forest/forest_topology.hpp"
+
+// Preludes
+using namespace parthenon::driver::prelude;
+
+using boundary_exchange::BoundaryExchangeDriver;
+
+Packages_t ProcessPackages(std::unique_ptr<ParameterInput> &pin);
+
+int main(int argc, char *argv[]) {
+  ParthenonManager pman;
+
+  pman.app_input->ProcessPackages = ProcessPackages;
+
+  // This is called on each mesh block whenever the mesh changes.
+  // pman.app_input->InitMeshBlockUserData = &calculate_pi::SetInOrOutBlock;
+
+  auto manager_status = pman.ParthenonInitEnv(argc, argv);
+  if (manager_status == ParthenonStatus::complete) {
+    pman.ParthenonFinalize();
+    return 0;
+  }
+  if (manager_status == ParthenonStatus::error) {
+    pman.ParthenonFinalize();
+    return 1;
+  }
+
+  std::unordered_map<uint64_t, std::shared_ptr<parthenon::forest::Node>> nodes;
+  nodes[0] = parthenon::forest::Node::create(0, {0.0, 0.0});
+  nodes[1] = parthenon::forest::Node::create(1, {1.0, 0.0});
+  nodes[2] = parthenon::forest::Node::create(2, {1.0, 1.0});
+  nodes[3] = parthenon::forest::Node::create(3, {0.0, 1.0});
+  nodes[4] = parthenon::forest::Node::create(4, {2.0, 0.0});
+  nodes[5] = parthenon::forest::Node::create(5, {2.0, 1.0});
+  nodes[6] = parthenon::forest::Node::create(6, {0.0, 2.0});
+  nodes[7] = parthenon::forest::Node::create(7, {1.0, 2.0});
+  nodes[8] = parthenon::forest::Node::create(8, {2.0, 2.0});
```

Comment:
Right now, nodes store a position in the x-y plane. This was convenient for making plots like the refinement plot shown above but the positions are unused otherwise. Probably in the future we don't actually want to associate nodes with positions since they are just there for specifying the topology of the mesh.

Reply:
added a comment explaining this